src/app/products/products.component.html

```diff
@@ -45,3 +45,7 @@ <h1></h1>
   <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
   <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
 </table>
+
+<button mat-button class="mat-button" color="custom" (click)="gocreate()">
+  Create new product
+</button>
```

Comment:
```suggestion
<a routerLink="/create-product">
<button mat-button class="mat-button" color="custom">
  Create new product
</button>
</a>
```

обрабатывать навигацию через клик ивенты на кнопке... сомнительное решение, когда у нас есть нативные ссылки с поддержкой ангуларовского роутера